service/matching/physical_task_queue_manager.go

```diff
@@ -544,8 +546,8 @@ func (c *physicalTaskQueueManagerImpl) ensureRegisteredInDeployment(
 	c.deploymentLock.Lock()
 	defer c.deploymentLock.Unlock()
 
-	if c.deploymentRegistered {
-		// already done
+	if c.deploymentRegistered || c.deploymentRegistrationNotPossible {
```

Comment:
we should store the registration error and return the same error to future pollers, in this case `ErrMaxTaskQueuesInDeployment`

Reply:
I agree - while we are at it, lets also return the repeated deployment error to be clear to the customer